pkg/aws/eni/node.go

```diff
@@ -271,24 +399,31 @@ func isSubnetAtPrefixCapacity(err error) bool {
 	return false
 }
 
-// AllocateIPs performs the ENI allocation oepration
-func (n *Node) AllocateIPs(ctx context.Context, a *ipam.AllocationAction) error {
+// AllocateIPs performs the ENI IP allocation operation
+func (n *Node) AllocateIPs(ctx context.Context, a *ipam.AllocationAction, family ipam.Family) error {
 	// Check if the interface to allocate on is prefix delegated
 	n.mutex.RLock()
-	isPrefixDelegated := n.node.Ops().IsPrefixDelegated()
+	isPrefixDelegated := n.node.Ops().IsPrefixDelegated(family)
 	n.mutex.RUnlock()
 
 	if isPrefixDelegated {
-		numPrefixes := ip.PrefixCeil(a.IPv4.AvailableForAllocation, option.ENIPDBlockSizeIPv4)
-		err := n.manager.api.AssignENIPrefixes(ctx, a.InterfaceID, int32(numPrefixes))
-		if !isSubnetAtPrefixCapacity(err) {
-			return err
+		if family == ipam.IPv6 {
+			numPrefixes := ip.PrefixCeil(a.IPv6.AvailableForAllocation, option.ENIPDBlockSizeIPv6)
+			if err := n.manager.api.AssignENIIPv6Prefixes(ctx, a.InterfaceID, int32(numPrefixes)); err != nil {
+				return err
```

Comment:
In contrast to the `isSubnetAtPrefixCapacity` logic below (which returns early if there is no error), this will not return early if there is no error. This means that if `AssignENIIPv6Prefixes` succeeded, we'll still call `n.manager.api.AssignPrivateIpAddresses` at the end of the function, which I think we don't want.